src/main/java/org/spongepowered/server/launch/VanillaServerMain.java

```diff
@@ -157,25 +237,27 @@ private static boolean downloadVerified(String remote, Path path) throws IOExcep
         System.out.println("Downloading " + name + "... This can take a while.");
         System.out.println(url);
         URLConnection con = url.openConnection();
-        MessageDigest md5 = MessageDigest.getInstance("MD5");
 
-        try (ReadableByteChannel source = Channels.newChannel(new DigestInputStream(con.getInputStream(), md5));
+        MessageDigest digest = MessageDigest.getInstance(digestType);
+
+        try (ReadableByteChannel source = Channels.newChannel(new DigestInputStream(con.getInputStream(), digest));
              FileChannel out = FileChannel.open(path, StandardOpenOption.CREATE_NEW, StandardOpenOption.WRITE)) {
             out.transferFrom(source, 0, Long.MAX_VALUE);
         }
 
-        String expected = getETag(con);
-        if (!expected.isEmpty()) {
-            String hash = toHexString(md5.digest());
-            if (hash.equals(expected)) {
-                System.out.println("Successfully downloaded " + name + " and verified checksum!");
-            } else {
-                Files.delete(path);
-                throw new IOException("Checksum verification failed: Expected " + expected + ", got " + hash);
-            }
+        if (expected == null) {
+            // Get the digest from the ETag header
+            expected = getETag(con);
```

Comment:
Can you remove the etag stuff entirely? The JSON will probably just fail to parse if it's invalid. No need to verify it additionally.